boilerplates/react-union-boilerplate-liferay-basic/hero-portlet/src/main/java/eu/reactunion/boilerplate/configuration/HeroConfigurationAction.java

```diff
@@ -0,0 +1,56 @@
+package eu.reactunion.boilerplate.configuration;
+
+import javax.portlet.ActionRequest;
+import javax.portlet.ActionResponse;
+import javax.portlet.PortletConfig;
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+
+import com.liferay.portal.kernel.module.configuration.ConfigurationProvider;
+import com.liferay.portal.kernel.portlet.ConfigurationAction;
+import com.liferay.portal.kernel.portlet.DefaultConfigurationAction;
+import com.liferay.portal.kernel.util.ParamUtil;
+import eu.reactunion.boilerplate.constants.HeroConstants;
+import eu.reactunion.boilerplate.constants.HeroPortletKeys;
+import org.osgi.service.component.annotations.Component;
+import org.osgi.service.component.annotations.ConfigurationPolicy;
+import org.osgi.service.component.annotations.Reference;
+
+/**
+ * The configuration action class enables Configuration item in the portlet's menu with gear icon.
+ * Configuration.jsp is used to render the UI.
+ *
+ * @author Roman Srom (roman.srom@lundegaard.eu)
+ */
+@Component(
+    configurationPid = HeroPortletKeys.CONFIGURATION,
+    configurationPolicy = ConfigurationPolicy.OPTIONAL,
+    immediate = true,
+    property = {
+        "javax.portlet.name=" + HeroPortletKeys.HERO,
+    },
+    service = ConfigurationAction.class
+)
+public class HeroConfigurationAction extends DefaultConfigurationAction {
+
+    @Reference
+    private ConfigurationProvider configurationProvider;
+
+    @Override
+    public void processAction(PortletConfig portletConfig, ActionRequest actionRequest, ActionResponse actionResponse) throws Exception {
+        String heading = ParamUtil.getString(actionRequest, HeroConstants.PARAM_HEADING);
```

Comment:
If no validation required it is possible to use preferences naming convetion (--pereference--heading) in aui input name and storing of preferences is done automatically.

Reply:
Implemented